drivers:adc:ltc2378: Add support for LTC2378-20

csarip opened the pull request:
## Pull Request Description

This PR adds initial driver support for the LTC2378-20 20-bit SAR ADC with project examples.

#### Driver Features Implemented:

- SPI Communication: 20-bit data acquisition with CNV/BUSY handshake protocol
- Input Modes: Unipolar (0 to VREF) and bipolar (±VREF) configuration support
- Data Conversion: Raw ADC reading and raw-to-microvolts conversion functions
- Power Management: Manual power-down control via CNV signal

#### Project Examples Added:
- Platform: Maxim MAX32666FTHR evaluation board with DC2135A board
- Test: Validates SPI communication, ADC conversion, unipolar/bipolar modes, and voltage scaling
- Hardware Validation: Tested on actual MAX32666FTHR hardware with DC2135A board

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have complied with the [Submission Checklist](http://analogdevicesinc.github.io/no-OS/contributing.html#submission-checklist)
- [x] I have performed a self-review of the changes
- [x] I have commented my code, at least hard-to-understand parts
- [x] I have build all projects affected by the changes in this PR
- [x] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [x] I have updated the documentation (wiki pages, ReadMe etc), if applies

amiclaus commented on drivers/adc/ltc2378/iio_ltc2378.c:
```diff
@@ -0,0 +1,201 @@
+			return iio_format_value(buf, len, IIO_VAL_INT, 1, vals);
+		}
+
+		int32_t voltage_uv;
```
variable declaration at start of function.

amiclaus reviewed (commented):
looks good overall, but the documentation issues ar valid. check for the warnings in the log file, those are treated as errors too.

jemfgeronimo reviewed (commented):
Some minor changes. The main change is to use the new example infrastructure.
Reference commits: https://github.com/analogdevicesinc/no-OS/commit/39e2d467b156dc0934a4c9a68fe27c65475d93e5, https://github.com/analogdevicesinc/no-OS/commit/81c90dc34300c6cb3f1b6d7ff3eb17a22e701039, https://github.com/analogdevicesinc/no-OS/commit/913998c7a0b94bb85edbe7a3136cafbced72d9f6
Reference PRs: https://github.com/analogdevicesinc/no-OS/pull/2795, https://github.com/analogdevicesinc/no-OS/pull/2421

jemfgeronimo reviewed (commented):
Some changes to fix the LinuxBuilds documentation errors.

jemfgeronimo commented on drivers/adc/ltc2378/ltc2378.c:
```diff
@@ -0,0 +1,248 @@
+		do {
+			no_os_gpio_get_value(dev->gpio_busy, &val);
+			no_os_udelay(1);
+		} while (val && --timeout);
```
Avoid using magic numbers. I suggest replace `1000` with a macro (e.g. `LTC2378_BUSY_TIMEOUT_US`) to make it configurable.

jemfgeronimo commented on projects/ltc2378/src/common/common_data.c:
```diff
@@ -0,0 +1,80 @@
+	.extra = GPIO_EXTRA
+};
+
+struct ltc2378_init_param ltc2378_ip = {
```
```suggestion
const struct ltc2378_init_param ltc2378_ip = {
```

jemfgeronimo commented on projects/ltc2378/src/platform/maxim/parameters.h:
```diff
@@ -0,0 +1,70 @@
+#if (TARGET_NUM == 32665)
+#define SPI_DEVICE_ID	1
+#define SPI_CS		0
+#endif
```
You can drop the `#if`-`#endif` lines if you are not supporting other platforms.

jemfgeronimo reviewed (commented):
Some minor changes.

csarip commented:
v4:

- Fixed IIO error handling by replacing sentinel values with proper error codes
- Removed magic numbers by adding descriptive macros
- Used no-OS utility functions for sign extension

jemfgeronimo reviewed (commented):
Just a minor change. Otherwise, LGTM.

buha commented on projects/ltc2378/README.rst:
```diff
@@ -0,0 +1,162 @@
+.. code-block:: bash
+
+	# Select the example you want by choosing basic or iio_example
+	EXAMPLE ?= basic
```
user should select either with 
```
make EXAMPLE=iio
```

or

```
export EXAMPLE=iio
make
```

whichever you want, but they shouldn't edit the Makefile

amiclaus commented on drivers/adc/ltc2378/ltc2378.h:
```diff
@@ -0,0 +1,122 @@
+/***************************************************************************//**
+ *   @file   ltc2378.h
+*   @brief  Header file of the LTC2378 Driver
```
inconsistent formatting.

amiclaus commented on drivers/adc/ltc2378/iio_ltc2378.h:
```diff
@@ -0,0 +1,56 @@
+#ifndef IIO_LTC2378_H
+#define IIO_LTC2378_H
+
+#include <stdbool.h>
```
is stdbool needed here?

amiclaus commented on drivers/adc/ltc2378/iio_ltc2378.h:
```diff
@@ -0,0 +1,55 @@
+
+struct ltc2378_iio_desc {
+	struct ltc2378_dev *ltc2378_dev;
+	struct iio_device *iio_dev;
```
where does this struct come from? i don't see any relevant include.

amiclaus reviewed (commented):
some more comments on my side, otherwise lgtm.

amiclaus commented:
/AzurePipelines run
